### PR TITLE
Chore/tt 525 run ctf tests against testnets

### PR DESCRIPTION
**Pull request by iljapavlovs**

### CODE REVIEW

File: integration-tests/docker/test_env/test_env_builder.go [hunk 1]
```diff
@@ -89,41 +98,64 @@ func (b *CLTestEnvBuilder) buildNewEnv(cfg *TestEnvConfig) (*CLClusterTestEnv, e
...
-			return te, err
+			return nil, err
 		}
 	}
 
+	bc, err := blockchain.NewEVMClientFromNetwork(networkConfig)
+	if err != nil {
+		return nil, err
+	}
+	// Get blockchain.EthereumClient as this is the only possible client for Geth
```
Comment (skudasov) [inline integration-tests/docker/test_env/test_env_builder.go:141]: @iljapavlovs This comment is no longer true, we should remove this code and use `blockchain.EVMClient`.
Currently, in both testnet and Geth cases we have `te.EthClient` as `*blockchain.EthereumClient` and it won't work for different networks.
 Please use `blockchain.EVMClient` in all the cases if possible, if using a concrete type is easier it should be used only when `hasGeth` flag is set.

File: integration-tests/smoke/vrfv2_test.go [hunk 2]
```diff
@@ -23,21 +23,24 @@ func TestVRFv2Basic(t *testing.T) {
 
 	env, err := test_env.NewCLTestEnvBuilder().
 		WithGeth().
-		WithMockServer(1).
 		WithCLNodes(1).
+		//WithCLNodeConfig(node.NewConfig(node.BaseConf,
+		//	node.WithOCR2(),
+		//	node.WithP2Pv2(),
+		//)).
```
Comment (skudasov) [inline integration-tests/smoke/vrfv2_test.go:30]: Remove it, please. We have examples in other tests.

File: integration-tests/types/config/node/core.go [hunk 3]
```diff
@@ -116,6 +119,33 @@ func SetDefaultSimulatedGeth(cfg *chainlink.Config, ws, http string) {
 	}
 }
 
+func SetChainConfig(cfg *chainlink.Config, wsUrls, httpUrls []string, chain blockchain.EVMNetwork) {
```
Comment (skudasov) [inline integration-tests/types/config/node/core.go:122]: Should we remove `SetDefaultSimulatedGeth` then? One method to configure both simulated and non-simulated.

**Review by skudasov — CHANGES_REQUESTED** [review]

File: integration-tests/docker/test_env/test_env_builder.go (hunk 1 above)
Comment (iljapavlovs) [reply integration-tests/docker/test_env/test_env_builder.go]: fixed

**Review by skudasov — APPROVED** [review]

File: integration-tests/docker/test_env/geth.go [hunk 4]
```diff
@@ -29,13 +28,10 @@ const (
 
 type Geth struct {
 	EnvComponent
-	ExternalHttpUrl  string
-	InternalHttpUrl  string
-	ExternalWsUrl    string
-	InternalWsUrl    string
-	EthClient        *blockchain.EthereumClient
```
Comment (lukaszcl) [inline integration-tests/docker/test_env/geth.go:36]: @iljapavlovs why did you remove EthClient, ContractDeployer etc from this struct? Those are initialized in the methods of `Geth` so why put them outside?

Comment (lukaszcl) [reply integration-tests/docker/test_env/geth.go:36]: cc @skudasov

File: integration-tests/docker/test_env/test_env.go [hunk 5]
```diff
@@ -64,10 +69,10 @@ func NewTestEnvFromCfg(cfg *TestEnvConfig) (*CLClusterTestEnv, error) {
 }
 
 func (te *CLClusterTestEnv) ParallelTransactions(enabled bool) {
-	te.Geth.EthClient.ParallelTransactions(enabled)
+	te.EVMClient.ParallelTransactions(enabled)
 }
 
-func (te *CLClusterTestEnv) StartGeth() error {
+func (te *CLClusterTestEnv) StartGeth() (blockchain.EVMNetwork, InternalDockerUrls, error) {
```
Comment (lukaszcl) [inline integration-tests/docker/test_env/test_env.go:75]: @iljapavlovs @skudasov why introduce `InternalDockerUrls` since we already have `Geth.InternalHttpUrl` and `Geth.WsUrl`?

File: integration-tests/types/config/node/core.go [hunk 6]
```diff
@@ -121,37 +138,14 @@ func SetDefaultSimulatedGeth(cfg *chainlink.Config, ws, http string, forwarders
 	}
 }
 
-func WithSimulatedEVM(httpUrl, wsUrl string) NodeConfigOpt {
```
Comment (lukaszcl) [inline integration-tests/types/config/node/core.go:124]: @iljapavlovs why did you remove this one?
